File: sistema-de-fechas/ruby/sistema-de-fechas.rb
require 'date'

class UtilidadesFecha
  def initialize
    @formato_fecha = '%d/%m/%Y %H:%M:%S.%L'
  end

  def de_texto_a_fecha(texto_fecha)
    partes_fecha = texto_fecha.scan(/\d+/).map(&:to_i)
    dia, mes, anio, hora, minutos, segundos, milisegundos = partes_fecha

    DateTime.new(anio, mes, dia, hora, minutos, segundos, milisegundos * 1000)
  end

  def de_fecha_a_texto(fecha)
    fecha.strftime(@formato_fecha)
  end
end

# Uso
utilidades_fecha = UtilidadesFecha.new

# Convertir de texto a fecha
fecha_desde_texto = utilidades_fecha.de_texto_a_fecha('01/02/2022 10:30:45.123')
puts "Fecha desde texto: #{fecha_desde_texto}"

# Convertir de fecha a texto
fecha_actual = DateTime.now
texto_desde_fecha = utilidades_fecha.de_fecha_a_texto(fecha_actual)
puts "Texto desde fecha: #{texto_desde_fecha}"

File: docs/sistema-de-fechas/js/sistema-de-fechas.js
class SistemaDeFechas {
  constructor() {
    this.formatoFecha = 'DD/MM/YYYY HH:mm:ss.SSS';
  }

  deTextoAFecha(textoFecha) {
    const partesFecha = textoFecha.split(/[\s/:.]+/);
    const [dia, mes, anio, hora, minutos, segundos, milisegundos] = partesFecha.map(Number);

    return new Date(anio, mes - 1, dia, hora, minutos, segundos, milisegundos || 0);
  }

  deFechaATexto(fecha) {
    const dia = fecha.getDate().toString().padStart(2, '0');
    const mes = (fecha.getMonth() + 1).toString().padStart(2, '0');
    const anio = fecha.getFullYear();
    const hora = fecha.getHours().toString().padStart(2, '0');
    const minutos = fecha.getMinutes().toString().padStart(2, '0');
    const segundos = fecha.getSeconds().toString().padStart(2, '0');
    const milisegundos = fecha.getMilliseconds().toString().padStart(3, '0');

    return `${dia}/${mes}/${anio} ${hora}:${minutos}:${segundos}.${milisegundos}`;
  }
}

// Uso
const SistemaDeFechas = new SistemaDeFechas();

// Convertir de texto a fecha
const fechaDesdeTexto = SistemaDeFechas.deTextoAFecha('01/02/2022 10:30:45.123');
console.log('Fecha desde texto:', fechaDesdeTexto);

// Convertir de fecha a texto
const textoDesdeFecha = SistemaDeFechas.deFechaATexto(new Date());
console.log('Texto desde fecha:', textoDesdeFecha);
